components/group/groupStorage.ts

```diff
@@ -0,0 +1,24 @@
+import { RegisteredGroup } from '../../lib/dto/GroupCollection';
+
+
+const groupStorageKey = "money_tracker_group_storage"
+export const getLocalStorageRegisteredGroups: () => RegisteredGroup[] = () => {
+    if (typeof window === 'undefined' || window.localStorage === undefined) {
+        return []
+    } else {
+        const currentStorageItemString = localStorage.getItem(groupStorageKey)
+        if (currentStorageItemString === null) {
+            window.localStorage.setItem(groupStorageKey, JSON.stringify([]))
+            return []
+        }
```

Comment:
Sometimes you reference `localStorage` through the window object, sometimes directly, It should generally be safe to omit `window.`, for global objects, unless there is a possibility for a conflict in the local scope. Anyways this looks weird, when both variants are used in the same function.

Reply:
I see little merit in initializing the local storage value to an empty array here, if we are returning an empty array when the value is `null` anyways